pkg/metricscollector/v1beta1/kfp-metricscollector/v1/metrics_loader.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The Kubeflow pipeline metrics collector KFPMetricParser parses the metrics file 
+# and returns an ObservationLog of the metrics specified.
+# Some documentation on the metrics collector file structure can be found here:
+# https://v0-6.kubeflow.org/docs/pipelines/sdk/pipelines-metrics/
+
+from datetime import datetime
+from logging import getLogger, StreamHandler, INFO
+import os
+from typing import List
+import json
+
+import rfc3339
+import api_pb2
+from pkg.metricscollector.v1beta1.common import const
+
+class KFPMetricParser:
+    def __init__(self, metric_names):
+        self.metric_names = metric_names
+
+    @staticmethod
+    def find_all_files(directory):
+        for root, dirs, files in os.walk(directory):
+            for f in files:
+                yield os.path.join(root, f)
+
+    def parse_metrics(self, fn: str) -> List[api_pb2.MetricLog]:
```

Comment:
```suggestion
    def parse_metrics(self, metric_file_path: str) -> List[api_pb2.MetricLog]:
```

Should we use `metric_file_path` as a name for more clarification?